scylla/src/transport/connection.rs

```diff
@@ -1578,22 +1583,29 @@ pub(crate) async fn open_named_connection(
         .first()
         .and_then(|p| p.parse::<u16>().ok());
 
+    // Parse nonstandard protocol extensions.
     let protocol_features = ProtocolFeatures::parse_from_supported(&supported.options);
 
-    let mut options = HashMap::new();
-    protocol_features.add_startup_options(&mut options);
-
+    // At the beginning, Connection assumes no sharding and no protocol extensions;
+    // now that we know them, let's turn them on in the driver.
     let features = ConnectionFeatures {
         shard_info,
         shard_aware_port,
         protocol_features,
     };
     connection.set_features(features);
 
+    /* Prepare options that the driver opts-in in STARTUP frame. */
+    let mut options = HashMap::new();
+    protocol_features.add_startup_options(&mut options);
+
+    // The only CQL protocol version supported by the driver.
```

Comment:
This change is not only adding comments. But this reordering seems harmelss, so let's ignore it.